static/js/tab_bar.js

```diff
@@ -138,6 +138,103 @@ exports.colorize_tab_bar = function () {
     }
 };
 
+function hide_stream_details() {
+    $(".stream_description").hide();
+    $(".number_of_users").hide();
+}
+
+function empty_nav_bar() {
+    $("#tab_list").addClass("hidden");
+    hide_stream_details();
+    $(".hash").hide();
+    $(".navbar-lock").hide();
+    $(".navbar-search").removeClass("expanded");
+}
+
+function render_stream_details(stream_term) {
+    $("#tab_list").removeClass("hidden");
+    var stream_name = stream_term.operand;
+    var stream  = stream_data.get_sub_by_name(stream_name);
+    if (stream === undefined) {
+        return;
+    }
+    $(".stream_description").text(stream.description);
+    $(".nav_bar_user_count").append("<i class=\"fa fa-user-o\"></i> ");
+    $(".nav_bar_user_count").append(stream.subscriber_count);
+    $(".stream_description").show();
+    $(".number_of_users").show();
+    if (stream.invite_only) {
+        $(".navbar-lock").show();
+    } else {
+        $(".hash").show();
+    }
```

Comment:
I would definitely do this function with a handlebars template that we render and then swap in as the new HTML for that area.  As a style/readability rule, we don't construct complex HTML inside JS code in this project.